conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineEntryPoint.kt

```diff
@@ -85,19 +90,33 @@ object GramineEntryPoint {
                     .getAccessibleMethod("setUserPythonScript", Path::class.java)
                     .execute(enclave, Paths.get(PYTHON_FILE))
         }
-        val env = createEnclaveEnvironment(enclaveClass, hostInterface)
+        validateEnclaveMode()
+        val env = GramineEnclaveEnvironment(enclaveClass, hostInterface, simulationMrSigner!!, enclaveMode)
         hostInterface.sanitiseExceptions = (env.enclaveMode == EnclaveMode.RELEASE)
         Enclave::class.java
             .getAccessibleMethod("initialise", EnclaveEnvironment::class.java)
             .execute(enclave, env)
     }
 
-    private fun createEnclaveEnvironment(enclaveClass: Class<*>, hostInterface: SocketEnclaveHostInterface): EnclaveEnvironment {
-        return if (isSimulation) {
-            GramineDirectEnclaveEnvironment(enclaveClass, hostInterface, signingKeyMeasurement!!)
-        } else {
-            System.err.println("Gramine SGX is not yet implemented.")
-            exitProcess(EXIT_ERR)
+    private fun validateEnclaveMode() {
+        when (enclaveMode) {
+            EnclaveMode.SIMULATION -> {
+                require(File(GRAMINE_MANIFEST).exists()) { "Gramine Direct $commonErrorMessage" }
+            }
+
+            EnclaveMode.DEBUG -> {
+                require(File(GRAMINE_SGX_MANIFEST).exists()) { "Gramine SGX $commonErrorMessage" }
+                require(isSgxDebug) { "Gramine SGX debug flag not enabled in $enclaveMode mode" }
+            }
+
+            EnclaveMode.RELEASE -> {
+                require(File(GRAMINE_SGX_MANIFEST).exists()) { "Gramine SGX $commonErrorMessage" }
+                require(!isSgxDebug) { "Gramine SGX debug flag not disabled in $enclaveMode mode" }
+            }
+
+            EnclaveMode.MOCK -> {
+                throw IllegalArgumentException("Developer error: MOCK mode not supported in Gramine")
```

Comment:
If we get here then I think it's a bug in the SDK. I don't think the user can recreate this, so remove "developer error"